lib/run/index.js

```diff
@@ -177,26 +180,29 @@ module.exports = function (options, callback) {
                     });
                 },
 
-                assertion: function (cursor, assertion) {
-                    !assertion && (assertion = {});
+                assertion: function (cursor, assertions) {
 
-                    // store the legacy assertion index
-                    assertion.index && (legacyAssertionIndices[cursor.ref] = assertion.index);
+                    _.forEach(assertions, function (assertion) {
+                        !assertion && (assertion = {});
 
-                    emitter.emit('assertion', (assertion.passed ? null : {
-                        name: _.get(assertion, 'error.name', ''),
-                        index: assertion.index,
-                        message: _.get(assertion, 'error.message', ''),
+                        // store the legacy assertion index
+                        assertion.index && (legacyAssertionIndices[cursor.ref] = assertion.index);
 
-                        stack: 'AssertionFailure: ' + _.get(assertion, 'error.message', '') + '\n' +
-                            '   at Object.eval sandbox-script.js:' + (assertion.index + 1) + ':' +
-                            ((cursor && cursor.position || 0) + 1) + ')'
-                    }), {
-                        cursor: cursor,
-                        assertion: assertion.name,
-                        skipped: assertion.skipped,
-                        error: assertion.error,
-                        item: run.resolveCursor(cursor)
+                        emitter.emit('assertion', (assertion.passed ? null : {
+                            name: _.get(assertion, 'error.name', ''),
+                            index: assertion.index,
+                            message: _.get(assertion, 'error.message', assertion.name || ''),
+
+                            stack: 'AssertionFailure: ' + _.get(assertion, 'error.message', '') + '\n' +
```

Comment:
@kamalaknn  We are not setting this to `AssertionError` as discussed?

Reply:
@shamasis the `message` is set to `AssertionError` from `runtime`. Only the `stack` is constructed here. Let me update the `stack` to match the `message`.